lib/tesla/middleware/telemetry.ex

```diff
@@ -18,6 +18,34 @@ if Code.ensure_loaded?(:telemetry) do
     end)
     ```
 
+    ## Options
+    - `:telemetry_prefix` - replaces default `[:tesla]` with desired Telemetry event prefix (see below)
+
+    ## Custom Telemetry Prefix
+
+    All events will use a `:telemetry_prefix` which defaults to `[:tesla]`.
+
+    You can customize events by providing your own `:telemetry_prefix` locally:
+
+    ```
+    defmodule MyClient do
+      use Tesla
+
+      plug Tesla.Middleware.Telemetry, telemetry_prefix: [:custom, :prefix]
+
+    end
+
+    :telemetry.attach("my-tesla-telemetry", [:custom, :prefix, :request, :stop], fn event, measurements, meta, config ->
+      # Do something with the event
+    end)
+    ```
+
+    You can configure `:telemetry_prefix` globally in your config, but if set the `:telemetry_prefix` option will override:
+
+    ```
+    config :tesla, Tesla.Middleware.Telemetry, telemetry_prefix: [:custom, :prefix]
```

Comment:
I don't think this is a good idea - it would override this all clients from all dependencies.

Since the use case comes from building API clients as packages I think it would be better to _hide_ tesla from end user and use this instead. 

```elixir
# config/config.exs
config :myclient, telemetry_prefix: [...]

# myclient.ex

defmodule Myclient do
  use Tesla
  @compile_config Application.compile_env(:myclient, :telemetry_prefix, [])
  plug Tesla.Middleware.Telemetry, prefix: Keyword.get(@compile_config, :prefix, [:myclient])
  # ...
end

# or top provide runtime configuration

defmodule Myclient do
  def new do
    config = Application.get_env(:myclient, :telemetry_prefix, [])
    middleware = [
      # ...
      {Tesla.Middleware.Telemetry, prefix: Keyword.get(@compile_config, :prefix, [:myclient])}
    ]

    Tesla.client(middleware, adapter)
  end
end
```

Reply:
Just updated. Let me know if that works.